src/state/opamSysInteract.ml

```diff
@@ -264,27 +285,72 @@ let packages_status packages =
-       >    CommonMark parsing and rendering library and program in C
-    *)
-    let re_installed = Re.(compile (seq [str "[installed]"; eol])) in
-    let re_pkg =
-      Re.(compile @@ seq
-            [ bol;
-              rep1 @@ alt [alnum; punct];
-              char '/';
-              group @@ rep1 @@ alt [alnum; punct];
-              space;
-            ])
+    let get_avail_w_virtuals () =
+      let package_provided str =
+        OpamSysPkg.of_string
+          (match OpamStd.String.cut_at str '=' with
+           | None -> str
+           | Some (p, _vc) -> p)
+      in
+      (* Output format:
+         >Repository      : core
+         >Name            : python
+         >Version         : 3.9.6-1
+         >Description     : Next generation of the python high-level scripting language
+         >Architecture    : x86_64
+         >URL             : https://www.python.org/
+         >Licenses        : custom
+         >Groups          : None
+         >Provides        : python3
+         >Depends On      : bzip2  expat  gdbm  libffi  libnsl  libxcrypt  openssl
+         >Optional Deps   : python-setuptools
+         >                  python-pip
+         >[...]
+      *)
+      (* Discard stderr to not have it pollute output. Plus, exit code is the
+         number of packages not found. *)
+      run_command ~discard_err:true "pacman" ["-Si"]
+      |> snd
+      |> List.fold_left (fun (avail, provides, latest) l ->
+          match OpamStd.String.split l ' ' with
+          | "Name"::":"::p::_ ->
+            p +++ avail, provides, Some (OpamSysPkg.of_string p)
+          | "Provides"::":"::"None"::[] -> avail, provides, latest
+          | "Provides"::":"::pkgs ->
+            let ps = OpamSysPkg.Set.of_list (List.map package_provided pkgs) in
+            let provides =
+              match latest with
+              | Some p -> OpamSysPkg.Map.add p ps provides
+              | None -> provides (* Bad pacman output ?? *)
+            in
+            ps ++ avail, provides, None
+          | _ -> avail, provides, latest)
+        (OpamSysPkg.Set.empty, OpamSysPkg.Map.empty, None)
+      |> (fun (a,p,_) -> a,p)
     in
-    let sys_installed, sys_available =
-      run_query_command "pacman" ["-Ss" ; names_re ()]
-      |> with_regexp_dbl ~re_installed ~re_pkg
+    let get_installed str_pkgs =
+      (* output:
+         >extra/cmake 3.17.1-1 [installed]
+         >    A cross-platform open-source make system
+         >extra/cmark 0.29.0-1
+         >    CommonMark parsing and rendering library and program in C
+      *)
+      let re_pkg =
+        Re.(compile @@ seq
+              [ bol;
+                rep1 @@ alt [alnum; punct];
+                char '/';
+                group @@ rep1 @@ alt [alnum; punct];
+                space;
+                rep1 any;
+                str "[installed]";
+                eol
+              ])
+      in
+      run_query_command "pacman" ["-Ss" ; names_re ~str_pkgs ()]
```

Comment:
why not using `pacman -Qi` ? It would avoid us to maintain two separate parsers for archlinux

Reply:
From what i know, `pacman -Qi` display only installed packages, so we can't use it to detect available ones.
Thinking about it would simplify the parser to use `pacman -Qs` instead.

Reply:
We already have `pacman -Si` for the available ones and `pacman -Ss` only displays installed packages too so using `pacman -Qi` would simplify everything

Reply:
We can keep `pacman -Ss` for now. It can always be changed later if needed.

Reply:
Changed to `-Qs`, faster, and as we want only installed packages status.

Reply:
mh, -Qs doesn't handle virtual packages though so if used it would have to be without arguments

Reply:
It doesn't return the provided package name, but recognises it
```
# pacman -Qs zeromq
local/zeromq 4.3.4-1
    Fast messaging system built on sockets. C and C++ bindings. aka 0MQ, ZMQ.
# pacman -Qs libzmq.so
local/zeromq 4.3.4-1
    Fast messaging system built on sockets. C and C++ bindings. aka 0MQ, ZMQ.
```

Reply:
oh, nevermind my testing had issues. Looks alright